pkg/backend/kubeflow/generator/generator.go

```diff
@@ -0,0 +1,13 @@
+package generator
+
+import (
+	pytorchv1alpha2 "github.com/kubeflow/pytorch-operator/pkg/apis/pytorch/v1alpha2"
+	tfv1alpha2 "github.com/kubeflow/tf-operator/pkg/apis/tensorflow/v1alpha2"
+
+	"github.com/caicloud/ciao/pkg/types"
+)
+
+type Interface interface {
```

Comment:
can u add docs to this interface and its methods?

Reply:
nit: there is some inconsistency here

```
pkg/backend/kubeflow/generator/generator.go
pkg/s2i/types.go
pkg/backend/types.go
```

both are meant to be interfaces for pluggable implementation but with different naming convention.